src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -420,64 +416,27 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
-        if is_shared_gallery_image_id(gallery_image_reference):
-            # the gallery image reference is shared gallery image id
-            from azure.cli.command_modules.vm._vm_utils import parse_shared_gallery_image_id
-            image_info = parse_shared_gallery_image_id(gallery_image_reference)
-            from ._client_factory import cf_shared_gallery_image
-            gallery_image_info = cf_shared_gallery_image(cmd.cli_ctx).get(
-                location=location, gallery_unique_name=image_info[0], gallery_image_name=image_info[1])
-        elif is_community_gallery_image_id(gallery_image_reference):
-            # the gallery image reference is community gallery image id
-            from azure.cli.command_modules.vm._vm_utils import parse_community_gallery_image_id
-            image_info = parse_community_gallery_image_id(gallery_image_reference)
-            from ._client_factory import cf_community_gallery_image
-            gallery_image_info = cf_community_gallery_image(cmd.cli_ctx).get(
-                location=location, public_gallery_name=image_info[0], gallery_image_name=image_info[1])
-        else:
-            # the gallery image reference is compute gallery image id
-            from msrestazure.tools import parse_resource_id
-            terms = parse_resource_id(gallery_image_reference)
-            gallery_image_reference_rg, gallery_name, gallery_image_definition = \
-                terms['resource_group'], terms['name'], terms['child_name_1']
-            client = _compute_client_factory(cmd.cli_ctx)
-            gallery_image_info = client.gallery_images.get(resource_group_name=gallery_image_reference_rg,
-                                                           gallery_name=gallery_name,
-                                                           gallery_image_name=gallery_image_definition)
-        if hasattr(gallery_image_info, 'hyper_v_generation'):
-            if gallery_image_info.hyper_v_generation == 'V1':
-                logger.warning(log_message)
-            elif gallery_image_info.hyper_v_generation == 'V2':
-                if hyper_v_generation == 'V1':
-                    logger.warning(log_message)
-                # will set default value of hyper_v_generation
-                if hasattr(gallery_image_info, 'features') and gallery_image_info.features \
-                        and 'SecurityType' in gallery_image_info.features \
-                        and gallery_image_info.features['SecurityType'] in \
-                        ['TrustedLaunchSupported', 'TrustedLaunch', 'TrustedLaunchAndConfidentialVmSupported'] \
-                        and security_type != 'TrustedLaunch':
-                    # will set default value of security_type
-                    logger.warning(log_message)
+        if not security_type:
+            security_type = 'Standard'
+        if security_type != 'TrustedLaunch':
+            logger.warning(UPGRADE_SECURITY_HINT)
```

Comment:
We no longer need to modify the default values based on the conditions of `hyper_v_generation` and security_type`, right?

Reply:
Hi @zhoxing-ms this change has been confirmed with Ajay:
https://github.com/Azure/azure-cli/issues/23614#issuecomment-1764369004